erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -116,37 +116,61 @@ def get_sms_args(self, notification_type=None, child_doctype=None, child_name=No
 
 		return sms_args
 
+def auto_schedule_next_project_templates():
+	for_date = getdate()
+	target_date = add_to_date(date=for_date, days=-1)
 
-def schedule_next_project_template(project_template, serial_no, args):
+	schedule_data = frappe.db.sql("""
+		select msd.project_template, ms.serial_no
+		from `tabMaintenance Schedule Detail` msd
+		inner join `tabMaintenance Schedule` ms on ms.name = msd.parent
+		where
+			msd.scheduled_date = %s
+			and ms.status = 'Active'
+			and ifnull(ms.serial_no, '') != ''
+			and ifnull(msd.project_template, '') != ''
+	""", target_date, as_dict=1)
+
+	for schedule in schedule_data:
+		schedule_next_project_template(schedule.project_template, schedule.serial_no, overwrite_existing=False)
+
+
+def schedule_next_project_template(project_template, serial_no, args=None, overwrite_existing=True):
 	if not project_template:
 		return
 
+	if args is None:
+		args = {}
+
 	args = frappe._dict(args)
-	if not args.reference_doctype or not args.reference_name:
-		frappe.throw(_("Invalid reference for Next Maintenance Schedule"))
 
 	template_details = frappe.db.get_value("Project Template", project_template, ["next_due_after", "next_project_template"], as_dict=1)
 	if not template_details or not template_details.next_due_after or not template_details.next_project_template:
 		return
 
 	doc = get_maintenance_schedule_doc(serial_no)
-	update_customer_and_contact(args, doc)
-
-	existing_templates = [d.get('project_template') for d in doc.get('schedules', []) if d.get('project_template')]
 
 	schedule = frappe._dict({
+		'project_template': template_details.next_project_template,
 		'reference_doctype': args.reference_doctype,
 		'reference_name': args.reference_name,
 		'reference_date': getdate(args.reference_date)
 	})
+	schedule.scheduled_date = schedule.reference_date + relativedelta(months=template_details.next_due_after)
 
-	if template_details.next_project_template not in existing_templates:
-		schedule.project_template = template_details.next_project_template
-		schedule.scheduled_date = schedule.reference_date + relativedelta(months=template_details.next_due_after)
-		schedule.scheduled_date = doc.adjust_scheduled_date_for_holiday(schedule.scheduled_date)
+	existing_row = [d for d in doc.get('schedules') if d.get("project_template") == template_details.next_project_template]
```

Comment:
We should make sure that the existing row is not already fulfilled or not way in the past